spec/requests/api/v1/registrations/create_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+describe 'POST api/v1/users/sign_up' do
+  subject { post user_registration_path, params:, as: :json }
+
+  let(:params) do
+    {
+      user: {
+        email: 'email@example.com',
+        password: 'password',
+        password_confirmation: 'password',
+        username: 'username',
+        first_name: 'first name',
+        last_name: 'last name'
+      }
+    }
+  end
+
+  context 'with correct params' do
+    before do
+      subject
+    end
+
+    it_behaves_like 'there must not be a Set-Cookie in Header'
+    it_behaves_like 'does not check authenticity token'
+
+    it 'returns success' do
+      expect(response).to be_successful
+    end
+
+    it 'returns the user' do
+      user = User.last
+      expect(json[:user][:id]).to eq(user.id)
+      expect(json[:user][:email]).to eq(params.dig(:user, :email))
+      expect(json[:user][:username]).to eq(params.dig(:user, :username))
+      expect(json[:user][:uid]).to eq(user.uid)
+      expect(json[:user][:provider]).to eq('email')
+      expect(json[:user][:first_name]).to eq(params.dig(:user, :first_name))
+      expect(json[:user][:last_name]).to eq(params.dig(:user, :last_name))
+    end
+
+    it 'returns a valid client and access token' do
+      user = User.last
+      token = response.header['access-token']
+      client = response.header['client']
+      expect(user.reload).to be_valid_token(token, client)
+    end
+  end
+
+  context 'with incorrect params' do
+    let(:params) do
+      {
+        user: {
+          email: 'email@example.com',
+          password: 'password',
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded password. This poses a serious security risk as it may allow unauthorized access to sensitive data or systems. Possible remediation approaches are:
	 1. Manage secret data using environment variables:
		 a) Identify the secret data and store that data in environment variables on your local and production server.
		 b) Replace the secrets with references to environment variables.
	 2. Use AWS Secrets Manager to store and access secrets:
		 a) Visit Secrets Manager on the AWS Management Console.
		 b) Choose the secret type on the console page and follow the instructions.
		 c) Use the suggested code samples to retrieve the secret.

To learn more about password management best practices, see [OWASP password storage cheat sheet](https://www.owasp.org/index.php/Password_Storage_Cheat_Sheet).